storage: print log message when blocked on snapshot semaphore

tbg opened the pull request:
We strongly suspect that this semaphore backs up the Raft snapshot
queue. Of course there's only so much the Raft snapshot queue can do when it
is asked to send ~100 Raft snapshots to a given node (this is just a lot
of data), but we also suspect that preemptive snapshots exacerbate the
problem since they are more aggressively rate limited. This logging will
allow confirming this in experiments.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/32712)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32712#-:-LSUX1A877JHZd4GxjJQ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/storage/store_snapshot.go, line 376 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32712#-LSUWsMa5idMrBmqHXDx:-LSUWsMb0NmlO6w26E6P:b-fyg1lp) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf1ed1981a5c97c3a92f74eac6596fed595f17fb/pkg/storage/store_snapshot.go#L376)):*
> ```Go
> 	}
> 
> 	const snapshotReservationWaitWarnThreshold = 13 * time.Second
> ```

That threshold is a bit random. Care to add a comment explaining the choice?


<!-- Sent from Reviewable.io -->

tbg committed:
storage: print log message when blocked on snapshot semaphore

We strongly suspect that this semaphore backs up the Raft snapshot
queue. Of course there's only so much the Raft snapshot queue can do when it
is asked to send ~100 Raft snapshots to a given node (this is just a lot
of data), but we also suspect that preemptive snapshots exacerbate the
problem since they are more aggressively rate limited. This logging will
allow confirming this in experiments.

Release note: None

tbg reviewed (commented):
bors r=petermattis

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32712#-:-LSUhnyUD-O1n2MpDkQY:blkd80i)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/storage/store_snapshot.go, line 376 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32712#-LSUWsMa5idMrBmqHXDx:-LSUhprF3NQ9w5Rpb_4B:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf1ed1981a5c97c3a92f74eac6596fed595f17fb/pkg/storage/store_snapshot.go#L376)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

That threshold is a bit random. Care to add a comment explaining the choice?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
